From b9c95f8a5b5cfaecffea6c1b1289df036429fb10 Mon Sep 17 00:00:00 2001
From: bakkeby <bakkeby@gmail.com>
Date: Tue, 22 Sep 2020 15:07:01 +0200
Subject: [PATCH] dwm_no_border_for_unselected_clients

---
 dwm.c | 13 ++++++++++++-
 1 file changed, 12 insertions(+), 1 deletion(-)

diff --git a/dwm.c b/dwm.c
index 664c527..1019457 100644
--- a/dwm.c
+++ b/dwm.c
@@ -803,6 +803,8 @@ focus(Client *c)
 		XDeleteProperty(dpy, root, netatom[NetActiveWindow]);
 	}
 	selmon->sel = c;
+	if (selmon->sel)
+		resizeclient(c, c->x, c->y, c->w, c->h);
 	drawbars();
 }
 
@@ -1282,7 +1284,13 @@ resizeclient(Client *c, int x, int y, int w, int h)
 	c->oldy = c->y; c->y = wc.y = y;
 	c->oldw = c->w; c->w = wc.width = w;
 	c->oldh = c->h; c->h = wc.height = h;
-	wc.border_width = c->bw;
+	if (c == selmon->sel)
+		wc.border_width = c->bw;
+	else {
+		wc.border_width = 0;
+		wc.x += c->bw;
+		wc.y += c->bw;
+	}
 	XConfigureWindow(dpy, c->win, CWX|CWY|CWWidth|CWHeight|CWBorderWidth, &wc);
 	configure(c);
 	XSync(dpy, False);
@@ -1760,6 +1768,9 @@ unfocus(Client *c, int setfocus)
 		XSetInputFocus(dpy, root, RevertToPointerRoot, CurrentTime);
 		XDeleteProperty(dpy, root, netatom[NetActiveWindow]);
 	}
+	if (c == selmon->sel)
+		selmon->sel = NULL;
+	resizeclient(c, c->x, c->y, c->w, c->h);
 }
 
 void
-- 
2.28.0

